crates/metadata/src/specs.rs

```diff
@@ -994,6 +1028,34 @@ where
     }
 }
 
+/// This is wrapper for [`TypeSpec`] which acts as a factory.
+/// The whole purpose of the factory is to replace the type of
+/// `Ok` variant of `Result` with `()` because constructors do not return
+/// any data upon successful instantiation.
+///
+/// # Important Note
+/// Only use this factory with constructors!
+pub struct TransformResult<T> {
+    marker: core::marker::PhantomData<fn() -> T>,
+}
+
+impl<O, E> TransformResult<core::result::Result<O, E>>
+where
+    E: TypeInfo + 'static,
+{
+    /// Produces [`TypeSpec`] for a `Result<(), E>`.
+    pub fn new_type_spec<S>(segments_opt: Option<S>) -> TypeSpec
+    where
+        S: IntoIterator<Item = &'static str>,
```

Comment:
My gut feeling is this should rather be for `<F as Form>::String` or it will break runtime metadata construction

Reply:
Nevermind. After giving this some more thought I think external consumers (e.g. other languages relying on runtime metadata construction) don't need this because they can produce the correct type first hand anyways.